include/alpaka/core/ApiCudaRt.hpp

```diff
@@ -0,0 +1,357 @@
+        using MemcpyKind_t = ::cudaMemcpyKind;
+        using Memcpy3DParms_t = ::cudaMemcpy3DParms;
+        using PitchedPtr_t = ::cudaPitchedPtr;
+
+        // constants
+        static constexpr auto success = ::cudaSuccess;
+        static constexpr auto errorNotReady = ::cudaErrorNotReady;
+        static constexpr auto errorHostMemoryAlreadyRegistered = ::cudaErrorHostMemoryAlreadyRegistered;
+        static constexpr auto errorHostMemoryNotRegistered = ::cudaErrorHostMemoryNotRegistered;
+        static constexpr auto errorUnknown = ::cudaErrorUnknown;
+
+        static constexpr auto eventDefault = cudaEventDefault;
+        static constexpr auto eventBlockingSync = cudaEventBlockingSync;
+        static constexpr auto eventDisableTiming = cudaEventDisableTiming;
+        static constexpr auto eventInterprocess = cudaEventInterprocess;
+
+        static constexpr auto hostRegisterDefault = cudaHostRegisterDefault;
+        static constexpr auto hostRegisterPortable = cudaHostRegisterPortable;
+        static constexpr auto hostRegisterMapped = cudaHostRegisterMapped;
+        static constexpr auto hostRegisterIoMemory = cudaHostRegisterIoMemory;
+
+        static constexpr auto memcpyDefault = ::cudaMemcpyDefault;
+        static constexpr auto memcpyDeviceToDevice = ::cudaMemcpyDeviceToDevice;
+        static constexpr auto memcpyDeviceToHost = ::cudaMemcpyDeviceToHost;
+        static constexpr auto memcpyHostToDevice = ::cudaMemcpyHostToDevice;
+
+        static constexpr auto streamDefault = cudaStreamDefault;
+        static constexpr auto streamNonBlocking = cudaStreamNonBlocking;
+
+        static constexpr auto deviceAttributeMaxBlockDimX = ::cudaDevAttrMaxBlockDimX;
+        static constexpr auto deviceAttributeMaxBlockDimY = ::cudaDevAttrMaxBlockDimY;
+        static constexpr auto deviceAttributeMaxBlockDimZ = ::cudaDevAttrMaxBlockDimZ;
+        static constexpr auto deviceAttributeMaxGridDimX = ::cudaDevAttrMaxGridDimX;
+        static constexpr auto deviceAttributeMaxGridDimY = ::cudaDevAttrMaxGridDimY;
+        static constexpr auto deviceAttributeMaxGridDimZ = ::cudaDevAttrMaxGridDimZ;
+        static constexpr auto deviceAttributeMaxSharedMemoryPerBlock = ::cudaDevAttrMaxSharedMemoryPerBlock;
+        static constexpr auto deviceAttributeMaxThreadsPerBlock = ::cudaDevAttrMaxThreadsPerBlock;
+        static constexpr auto deviceAttributeMultiprocessorCount = ::cudaDevAttrMultiProcessorCount;
+
+        static constexpr auto limitPrintfFifoSize = ::cudaLimitPrintfFifoSize;
+
+        // runtime API
+        static inline cudaError_t deviceGetAttribute(int* value, enum cudaDeviceAttr attr, int device)
+        {
+            return ::cudaDeviceGetAttribute(value, attr, device);
+        }
+
+        static inline cudaError_t deviceGetLimit(size_t* pValue, enum cudaLimit limit)
+        {
+            return ::cudaDeviceGetLimit(pValue, limit);
+        }
+
+        static inline cudaError_t deviceReset(void)
```

Comment:
```suggestion
        static inline cudaError_t deviceReset()
```
I personally dislike this relic from C. But this is not blocking this PR.